File: todo-app-backend/src/main/java/de/htw/berlin/todo_app_backend/domain/ToDoEntry.java
package de.htw.berlin.todo_app_backend.domain;

import de.htw.berlin.todo_app_backend.enums.Priority;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
public class ToDoEntry {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotNull
    @Size(max = 25)
    private String title;
    @Size(max = 200)
    private String description;
    private boolean done;
    private LocalDate dueDate;
    @Enumerated(EnumType.STRING)
    private Priority priority;
    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    void onCreate() {
        if (createdAt == null) {
            createdAt = LocalDateTime.now();
        }
    }
}

File: todo-app-ui/src/domain/todo-entry.ts
import type {Category} from "@/domain/category.ts";
import type {PriorityEnum} from "@/domain/priority.enum.ts";

export interface TodoEntry {
    id?: number;
    title?: string;
    description?: string;
    done?: boolean;
    createdAt?: string;
    dueDate?: string;
    priority?: PriorityEnum;
    category?: Category | null;
}
